plugin.py

```diff
@@ -86,10 +86,28 @@ def install_or_update(cls) -> None:
     def markdown_language_id_to_st_syntax_map(cls) -> Optional["MarkdownLangMap"]:
         return {"python": (("python", "py"), ("LSP-pyright/syntaxes/pyright",))}
 
+    def on_server_response_async(self, method: str, response: Response) -> None:
+        if method == "textDocument/hover" and isinstance(response.result, dict):
+            contents = response.result.get("contents")
+            if not isinstance(contents, dict) or not contents.get("kind") == "markdown":
+                return
+            response.result["contents"]["value"] = self.patch_markdown_content(contents["value"])
+
     # -------------- #
     # custom methods #
     # -------------- #
 
+    def patch_markdown_content(self, content: str) -> str:
+        # Add another linebreak before horizontal rule following fenced code block
+        content = re.sub("```\n---", "```\n\n---", content)
+        # Add markup for some common field name conventions in function docstring
+        content = re.sub(
+            r"\n:param[ ]+([\w\\]+):", lambda m: "\n__Param:__ `{}`".format(m.group(1).replace("\\_", "_")), content
+        )
+        content = re.sub(r"\n:raises[ ]+(\w+):", r"\n__Raises:__ `\1`", content)
+        content = re.sub(r"\n:returns?:", r"\n__Returns:__", content)
```

Comment:
I would like it to support Sphinx style too!

```py
def foo(arg1: int, arg2: str) -> bool:
    """
    { function_description }

    :param      arg1:  The argument 1
    :type       arg1:  int
    :param      arg2:  The argument 2
    :type       arg2:  str

    :returns:   { description_of_the_return_value }
    :rtype:     bool
    """
```

Reply:
Let's first try to suggest this to pyright or even help implementing in pyright. Including a parser for whole new syntax in LSP-pyright seems a bit excessive.

Reply:
> Let's first try to suggest this to pyright or even help implementing in pyright. Including a parser for whole new syntax in LSP-pyright seems a bit excessive.

According to https://github.com/microsoft/pylance-release/issues/1113#issuecomment-954122850 it is intended to not apply any formatting to the Sphinx-style (reST) field lists in hover popups from the Pylance extension for VSCode, and render them as raw text instead. I also found https://github.com/microsoft/pylance-release/issues/2251 and a few more similar requests, but it appears that there is no real action taken or deemed to be necessary about this.

Reply:
They are not strictly against improving it:

> Feel free to file a new issue if you have an idea of what you'd prefer instead.

They base the request on interest. Good start would be creating a specific issue for formatting those params better
(not saying **you** should :).

Reply:
> Good start would be creating a specific issue for formatting those params better

There are already plenty of issue reports about this. If it gets created in the microsoft/pyright repository, it will be transfered to microsoft/pylance-release because it isn't a core type checking feature and is related instead to language server features (https://github.com/microsoft/pylance-release/issues/2294#issuecomment-1022846247).

After that, it will be converted to a discussion, so that it can easily be put on ice / ignored.
Besides the two links I've already referenced in the previous comment, there are some more duplicates about this:
* https://github.com/microsoft/pylance-release/issues/3185
* https://github.com/microsoft/pylance-release/discussions/1130
* ...

I don't really see a reason why we should create another one of it.

Reply:
Yeah ok, seems like the first one is exactly what we needed.

Reply:
Thumbs up on https://github.com/microsoft/pylance-release/discussions/3186 wouldn't hurt. This might be how they gouge interest.

Reply:
It looks like they are open to improving this (see the link above) but would want to see a more formal enhancement suggestion first. I suppose it would be a short comment describing the problem, motivation for changing and potentially suggested solution. I could write something but you are much better at english then me and you already did a lot of work on that. :)

Reply:
In the mean time I'm fine with merging this but ideally it would wait for the LSP fix to be released first.

Reply:
Well, I think the problem and motivation is pretty clear already from the screenshot in the opening comment of that discussion, and from all the previous enhancement requests.
I assume they demand some kind of draft how the result should look and how exactly you would implement it. For example the simple workaround with the regex-substitution I use here in this PR, and resulting in the __Params:__ prefixes on each of the parameter lines, wouldn't be very satisfying I guess. Instead, you would need to collect all of the parameters first and then create an unordered list in Markdown with the parameters, for example, maybe also add a dash between parameter name and parameter description (dependent whether description exists or not), etc.
But since pyright is implemented in TypeScript, and I have absolutely zero experience with TypeScript (besides from looking at the LSP specs), I guess I wouldn't want to spend the time to figure out an implementation and the details. If you want to invest the time for an implementation, I wouldn't complain of course, but for me personally the simple approach from this PR is sufficient for now, and it was quick and easy to write just a handful lines of Python code.

> but you are much better at english then me

disagree 😄